server/lib/budget.js

```diff
@@ -695,59 +695,109 @@ export async function sumCollectivesTransactions(
   return totals;
 }
 
-export async function getYearlyIncome(collective) {
+export async function getYearlyIncome(collectiveIds) {
+  const ids = Array.isArray(collectiveIds) ? collectiveIds : [collectiveIds];
+
   // Three cases:
   // 1) All active monthly subscriptions. Multiply by 12
   // 2) All one-time and yearly subscriptions
   // 3) All inactive monthly subscriptions that have contributed in the past
 
-  // TODO: support netAmountInHostCurrency
-  const result = await sequelize.query(
+  const yearlyIncomeQuery = await sequelize.query(
     `
-      SELECT
-        (
-          SELECT COALESCE(SUM(o."totalAmount"), 0) * 12
+      WITH active_subscriptions as (
+        SELECT 
+            o."CollectiveId",
+            o.currency, 
+            (COALESCE(SUM(o."totalAmount") FILTER(WHERE s.interval = 'month'), 0) * 12 +
+            COALESCE(SUM(o."totalAmount") FILTER(WHERE s.interval = 'year'), 0)) as amount
           FROM "Orders" o
           INNER JOIN "Subscriptions" s ON o."SubscriptionId" = s.id
-          WHERE o."CollectiveId" = :CollectiveId
           AND o."deletedAt" IS NULL
           AND s."deletedAt" IS NULL
           AND s."isActive" IS TRUE
-          AND s.interval = 'month'
-        )
-        +
-        (
-          SELECT COALESCE(SUM(o."totalAmount"), 0)
-          FROM "Orders" o
-          INNER JOIN "Subscriptions" s ON o."SubscriptionId" = s.id
-          WHERE o."CollectiveId" = :CollectiveId
-          AND o."deletedAt" IS NULL
-          AND s."deletedAt" IS NULL
-          AND s."isActive" IS TRUE
-          AND s.interval = 'year'
-        )
-        +
-        (
-          SELECT COALESCE(SUM(t."netAmountInCollectiveCurrency"), 0)
+          GROUP BY o."CollectiveId", o.currency
+      ),
+      last_twelve_months_one_time_contributions as (
+          SELECT 
+            t."CollectiveId",
+            t."hostCurrency" as currency, 
+            SUM(t."amountInHostCurrency") +
+              SUM(coalesce(t."platformFeeInHostCurrency", 0)) +
+              SUM(coalesce(t."hostFeeInHostCurrency", 0)) +
+              SUM(coalesce(t."paymentProcessorFeeInHostCurrency", 0)) +
+              SUM(coalesce(t."taxAmount" * t."hostCurrencyFxRate", 0)) "amount"
           FROM "Transactions" t
-          LEFT JOIN "Orders" d ON t."OrderId" = d.id
-          LEFT JOIN "Subscriptions" s ON d."SubscriptionId" = s.id
-          WHERE t."CollectiveId" = :CollectiveId
-          AND t."RefundTransactionId" IS NULL
+          LEFT JOIN "Orders" o ON t."OrderId" = o.id
+          LEFT JOIN "Subscriptions" s ON s.id = o."SubscriptionId"
```

Comment:
If we want to fetch one time contributions, no need to join Subscriptions.

If we do, it's different than what is implied by "last_twelve_months_one_time_contributions".

Reply:
I think i picked a bad alias for that query. The original one does join with subscriptions, it sums up past recurring contributions that are not active anymore.